include/zephyr/net/net_ip.h

```diff
@@ -730,7 +730,7 @@ static inline bool net_ipv4_is_addr_mcast(const struct in_addr *addr)
  */
 static inline bool net_ipv4_is_ll_addr(const struct in_addr *addr)
 {
-	return (ntohl(UNALIGNED_GET(&addr->s_addr)) & 0xA9FE0000) == 0xA9FE0000;
+	return (ntohl(UNALIGNED_GET(&addr->s_addr)) & 0xFFFF0000) == 0xA9FE0000;
```

Comment:
Hmm, what is the actual issue here, why the original check would not work?

Reply:
Doing bitwise AND with `0xA9FE0000` could clear the bits (from the subnet part) that were orignally set in the provided address.

Take a IPv4 boradcast address as an example (`255.255.255.255`) - the original approach would modify the subnet part of the address, leading to false match:
`0xFFFFFFFF & 0xA9FE0000` gives `0xA9FE0000`, so the function would indicate it was the LL address. The same issue would apply to any address, that had a match on the bits set in `0xA9FE0000`.

Reply:
Indeed, good point.